wasm/src/varint.rs

```diff
@@ -0,0 +1,99 @@
+use crate::error::Error;
+
+const SEGMENT_BITS: u8 = 0x7F;
+const CONTINUE_BIT: u8 = 0x80;
+
+pub fn decode_u32<R: std::io::Read>(reader: &mut R) -> Result<(u32, u8), Error> {
```

Comment:
Nit: It might be easier to read and understand this with a struct like
```rust
struct DecodedU32Result {
 value: u32,
 bytes_read: u8
}
```
Although... technically there's a mathematical relationship between those two values, so you could logarithmically check the u32 result to reconstruct the byte count consumed?

Or, is this byte sequence valid: `[0x80, 0x80, 0x80, 0x80, 0x00]`?

Reply:
That is a valid technically. but basically never used because it results in `0`